python/cudf/cudf/tests/test_dataframe.py

```diff
@@ -4781,6 +4781,152 @@ def test_rowwise_ops_nullable_int_dtypes(op, expected):
+            ),
+            "t3": gd.Series(
+                ["1960-08-31 06:00:00", "2030-08-02 10:00:00"], dtype="<M8[s]"
+            ),
+        },
+        {
+            "t1": gd.Series(
+                ["2020-08-01 09:00:00", "1920-05-01 10:30:00"], dtype="<M8[ms]"
+            ),
+            "t2": gd.Series(
+                ["1940-08-31 06:00:00", "2020-08-02 10:00:00"], dtype="<M8[us]"
+            ),
+        },
+        {
+            "t1": gd.Series(
+                ["2020-08-01 09:00:00", "1920-05-01 10:30:00"], dtype="<M8[ms]"
+            ),
+            "t2": gd.Series(
+                ["1940-08-31 06:00:00", "2020-08-02 10:00:00"], dtype="<M8[ms]"
+            ),
+            "i1": gd.Series([1001, 2002], dtype="int64"),
+        },
+        {
+            "t1": gd.Series(
+                ["2020-08-01 09:00:00", "1920-05-01 10:30:00"], dtype="<M8[ms]"
+            ),
+            "t2": gd.Series(["1940-08-31 06:00:00", None], dtype="<M8[ms]"),
+            "i1": [1001, 2002],
+        },
+    ],
+)
+@pytest.mark.parametrize("op", ["max", "min"])
+@pytest.mark.parametrize("skipna", [True, False])
+def test_rowwise_ops_datetime_dtypes(data, op, skipna):
+
+    gdf = gd.DataFrame(data)
+    gdf_copy = gdf.copy(deep=True)
+
+    pdf = gdf.to_pandas()
+
+    got = getattr(gdf, op)(axis=1, skipna=skipna)
+    expected = getattr(pdf, op)(axis=1, skipna=skipna)
+
+    assert_eq(got, expected)
+    assert_eq(gdf, gdf_copy)
+
+
+@pytest.mark.parametrize(
+    "data,op,skipna",
+    [
+        (
+            {
+                "t1": gd.Series(
+                    ["2020-08-01 09:00:00", "1920-05-01 10:30:00"],
+                    dtype="<M8[ms]",
+                ),
+                "t2": gd.Series(
+                    ["1940-08-31 06:00:00", None], dtype="<M8[ms]"
+                ),
+            },
+            "max",
+            True,
+        ),
+        (
+            {
+                "t1": gd.Series(
+                    ["2020-08-01 09:00:00", "1920-05-01 10:30:00"],
+                    dtype="<M8[ms]",
+                ),
+                "t2": gd.Series(
+                    ["1940-08-31 06:00:00", None], dtype="<M8[ms]"
+                ),
+            },
+            "min",
+            False,
+        ),
+        (
+            {
+                "t1": gd.Series(
+                    ["2020-08-01 09:00:00", "1920-05-01 10:30:00"],
+                    dtype="<M8[ms]",
+                ),
+                "t2": gd.Series(
+                    ["1940-08-31 06:00:00", None], dtype="<M8[ms]"
+                ),
+            },
+            "min",
+            True,
+        ),
+    ],
+)
+def test_rowwise_ops_datetime_dtypes_2(data, op, skipna):
+
+    gdf = gd.DataFrame(data)
+    gdf_copy = gdf.copy(deep=True)
+
+    pdf = gdf.to_pandas()
+
+    got = getattr(gdf, op)(axis=1, skipna=skipna)
+    expected = getattr(pdf, op)(axis=1, skipna=skipna)
+
+    assert_eq(got, expected)
+    assert_eq(gdf, gdf_copy)
```

Comment:
Why are we checking that `gdf` is equal to `gdf_copy` here? Is there concerns about mutability?

Reply:
I think so. In `cudf` the `datetime64` values are stored w.r.t. their base time unit. For example:
```
>>> s = gd.Series(["2020-05-01 06:00:00"], dtype="<M8[ms]")
>>> s.astype("int64")
0    1588312800000
dtype: int64
>>> s2 = gd.Series(["2020-05-01 06:00:00"], dtype="<M8[ns]")
>>> s2.astype("int64")
0    1588312800000000000
dtype: int64
```
So in order to do row-wise in cupy (which natively does not support `datetime64` types), we will need to convert the values to the same base time unit first. This check is to make sure the values are converted back after the row-wise ops.

Reply:
Sounds good, thanks for the explanation. Can you add a comment either in one of the tests or above the test explaining this? There's lots of implementation detail specific information in there 😄

Reply:
"Comment to explain for **why** we are doing this."